numpy/random/generator.pyx

```diff
@@ -3475,21 +3492,35 @@ cdef class Generator:
 
         # GH10839, ensure double to make tol meaningful
         cov = cov.astype(np.double)
-        (u, s, v) = svd(cov)
+        if method == 'svd':
+            from numpy.dual import svd
+            (u, s, vh) = svd(cov)
+        elif method == 'eigh':
+            from numpy.dual import eigh
+            (s, u)  = eigh(cov)
+        else:
+            from numpy.dual import cholesky
+            l = cholesky(cov)
 
-        if check_valid != 'ignore':
+        if check_valid != 'ignore' and method != 'cholesky':
             if check_valid != 'warn' and check_valid != 'raise':
-                raise ValueError("check_valid must equal 'warn', 'raise', or 'ignore'")
-
-            psd = np.allclose(np.dot(v.T * s, v), cov, rtol=tol, atol=tol)
+                raise ValueError(
+                    "check_valid must equal 'warn', 'raise', or 'ignore'")
+            if method == 'svd':
+                psd = np.allclose(np.dot(vh.T * s, vh), cov, rtol=tol, atol=tol)
+            else:
+                psd = not np.any(s < -tol)
             if not psd:
                 if check_valid == 'warn':
                     warnings.warn("covariance is not positive-semidefinite.",
                                   RuntimeWarning)
                 else:
                     raise ValueError("covariance is not positive-semidefinite.")
 
-        x = np.dot(x, np.sqrt(s)[:, None] * v)
+        if method == 'cholesky':
+            x = np.dot(x, l)
+        else:
+            x = np.dot(x, u * np.sqrt(s))
```

Comment:
Why did this change to using `u` instead of `v` for the svd case? Does the difference matter?

If it was for convenience, I think my comment addresses how to get that without a stream change.

Note that the check on line 2510 is looking at `vh` - it's not clear to me that checking `vh` then using `u` is valid.

Reply:
For symmetric matrices v == u, so vh = u.T. I'll change this to u instead for consistency.

Reply:
Same applies here, should use `np.conj(u.T)`.

Reply:
@eric-wieser not sure if the latest commit is the requested change you had in mind?